bodhi/server/util.py

```diff
@@ -684,3 +684,35 @@ def sort_severity(value):
     }
 
     return value_map.get(value, 99)
+
+
+def infer_content_class(base, build):
+    """ Given a base class and a build from koji, identify and return the child
+    class associated with the appropriate ContentType.
+
+    For example, given the Package base class and a normal koji build, return
+    the RpmPackage model class. Or, given the Build base class and a container
+    build, return the ContainerBuild model class.
+
+    Arguments:
+    base -- A base model class.
+    build -- A dict of information from the build system (koji).
+    """
+
+    # Import this here to avoid circular import
+    from bodhi.server.models import ContentType
+
+    if 'extra' not in build:
+        raise KeyError("No extra block found for build %r" % build)
+
+    if build['extra'] is None:
+        identity = ContentType.rpm
+    elif 'module' in build['extra'].get('typeinfo', {}):
+        identity = ContentType.module
+    elif 'container_koji_task_id' in build['extra']:
+        # TODO - implement containers as yet another content type someday.
+        raise NotImplementedError("Inferred type 'container' is unhandled.")
```

Comment:
> So what happens when someone inputs container build NVR after this? Will they get some semi-verbose error?

Yeah, this should be improved.  They'll get a 500 error.

Perhaps they should get a [501 error](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#5xx_Server_error)?